components/brave_wallet_ui/utils/asset-utils.ts

```diff
@@ -14,20 +18,51 @@ export const getUniqueAssets = (assets: BraveWallet.BlockchainToken[]) => {
   })
 }
 
-export const isSelectedAssetInAssetOptions = (selectedAsset: BraveWallet.BlockchainToken, assetOptions: BraveWallet.BlockchainToken[]) => {
+export const isSelectedAssetInAssetOptions = (
+  selectedAsset: BraveWallet.BlockchainToken,
+  assetOptions: BraveWallet.BlockchainToken[]
+) => {
   return assetOptions.findIndex(asset => {
     return asset.contractAddress.toLowerCase() === selectedAsset?.contractAddress.toLowerCase() &&
       asset.chainId === selectedAsset.chainId &&
       asset.symbol.toLowerCase() === selectedAsset.symbol.toLowerCase()
   }) !== -1
 }
 
+export const getWyreAssetSymbol = (asset: BraveWallet.BlockchainToken) => {
+  if (
+    !asset.contractAddress || // gas coins ok
+    asset.chainId === BraveWallet.MAINNET_CHAIN_ID // ETH-ERC coins ok
+  ) {
+    return asset.symbol
+  }
+
+  // format non-ethereum EVM token symbols for Wyre
+  const prefix = getWyreNetworkPrefix(asset.chainId)
+  return prefix ? `${prefix}${asset.symbol.toUpperCase()}` : asset.symbol
+}
+
 export const getRampAssetSymbol = (asset: BraveWallet.BlockchainToken) => {
   if (asset.symbol.toUpperCase() === 'BAT' && asset.chainId === BraveWallet.MAINNET_CHAIN_ID) {
     // BAT is the only token on Ethereum Mainnet with a prefix on Ramp.Network
     return 'ETH_BAT'
   }
 
+  if (asset.chainId === BraveWallet.AVALANCHE_MAINNET_CHAIN_ID && asset.contractAddress === '') {
+    return asset.symbol // AVAX native token has no prefix
+  }
+
   const rampNetworkPrefix = getRampNetworkPrefix(asset.chainId)
   return rampNetworkPrefix !== '' ? `${rampNetworkPrefix}_${asset.symbol.toUpperCase()}` : asset.symbol
 }
+
+export const addLogoToToken = (token: BraveWallet.BlockchainToken) => {
+  return {
+    ...token,
+    logo: token.logo?.startsWith('ipfs://')
```

Comment:
Can we consolidate this logic into a helper function. I've seen it reused a few times across #13370 and #12510

Reply:
This logic was moved into the utils folder for this purpose. This is now exported for usage as a helper function

Reply:
Sorry I meant the logo specific logic

Reply:
Resolving this one for now, this is a small refactor just to prevent code reuse more than anything so not something that would block this